src/quark-core/src/EIP1271MultiSigner.sol

```diff
@@ -0,0 +1,158 @@
+    bytes4 internal constant EIP_1271_MAGIC_VALUE = 0x1626ba7e;
+
+    /// @notice The number of signatures required
+    uint256 public immutable requiredSignatures;
+
+    /// @notice The address of signer #0, or address(0) if unset.
+    address public immutable signer0;
+
+    /// @notice The address of signer #1, or address(0) if unset.
+    address public immutable signer1;
+
+    /// @notice The address of signer #2, or address(0) if unset.
+    address public immutable signer2;
+
+    /// @notice The address of signer #3, or address(0) if unset.
+    address public immutable signer3;
+
+    /// @notice The address of signer #4, or address(0) if unset.
+    address public immutable signer4;
+
+    /// @notice The address of signer #5, or address(0) if unset.
+    address public immutable signer5;
+
+    /// @notice The address of signer #6, or address(0) if unset.
+    address public immutable signer6;
+
+    /// @notice The address of signer #7, or address(0) if unset.
+    address public immutable signer7;
+
+    /// @notice The address of signer #8, or address(0) if unset.
+    address public immutable signer8;
+
+    /// @notice The address of signer #9, or address(0) if unset.
+    address public immutable signer9;
+
+    /**
+     * @notice Construct a new EIP1271MultiSigner
+     * @dev Signers must be passed in in sort order (lowest address first).
+     * @param requiredSignatures_ The required number of signers for a signature to be valid.
+     * @param signer0_ The address of signer #0, or address(0) if unset.
+     * @param signer1_ The address of signer #1, or address(0) if unset.
+     * @param signer2_ The address of signer #2, or address(0) if unset.
+     * @param signer3_ The address of signer #3, or address(0) if unset.
+     * @param signer4_ The address of signer #4, or address(0) if unset.
+     * @param signer5_ The address of signer #5, or address(0) if unset.
+     * @param signer6_ The address of signer #6, or address(0) if unset.
+     * @param signer7_ The address of signer #7, or address(0) if unset.
+     * @param signer8_ The address of signer #8, or address(0) if unset.
+     * @param signer9_ The address of signer #9, or address(0) if unset.
+     */
+    constructor(
+        uint256 requiredSignatures_,
+        address signer0_,
+        address signer1_,
+        address signer2_,
+        address signer3_,
+        address signer4_,
+        address signer5_,
+        address signer6_,
+        address signer7_,
+        address signer8_,
+        address signer9_
+    ) {
+        requiredSignatures = requiredSignatures_;
+        signer0 = signer0_;
+        signer1 = signer1_;
+        signer2 = signer2_;
+        signer3 = signer3_;
+        signer4 = signer4_;
+        signer5 = signer5_;
+        signer6 = signer6_;
+        signer7 = signer7_;
+        signer8 = signer8_;
+        signer9 = signer9_;
+
+        require(signer1_ == address(0) || signer1_ > signer0_, MisplacedSigner());
```

Comment:
as discussed we should probably flip the order so that zeros are trailing signers